nodes/core/core/89-delay.html

```diff
@@ -129,13 +129,13 @@
         defaults: {
             name: {value:""},
             pauseType: {value:"delay", required:true},
-            timeout: {value:"5", required:true, validate:RED.validators.number()},
+            timeout: {value:"5", required:true, validate:function(v) { return RED.validators.number(v) && (v >= 0); }},
             timeoutUnits: {value:"seconds"},
-            rate: {value:"1", required:true, validate:RED.validators.number()},
+            rate: {value:"1", required:true, validate:function(v) { return RED.validators.number(v) && (v >= 0); }},
             nbRateUnits: {value:"1", required:false, validate:RED.validators.regex(/\d+|/)},
             rateUnits: {value: "second"},
-            randomFirst: {value:"1", required:true, validate:RED.validators.number()},
-            randomLast: {value:"5", required:true, validate:RED.validators.number()},
+            randomFirst: {value:"1", required:true, validate:function(v) { return RED.validators.number(v) && (v >= 0); }},
+            randomLast: {value:"5", required:true, validate:function(v) { return RED.validators.number(v) && (v >= 0); }},
             randomUnits: {value: "seconds"},
```

Comment:
Is it possible that old deploys will have some of these properties still `undefined`? Should the validate functions return true if they are undefined so the nodes don't needlessly flag errors?